restaurants/views.py

```diff
@@ -1,3 +1,52 @@
-from django.shortcuts import render
+import json, re, bcrypt, jwt
 
-# Create your views here.
+from django.views         import View
+from django.http          import JsonResponse
+from django.db.models     import Q, Avg
+
+
+from reviews.models       import Review, ReviewImage
+from users.models         import User, WishList
+from restaurants.models   import Category, Location, Menu, Restaurant
+
+class SearchView(View):
+    def get(self, request):
+        search = request.GET.get('search', None)
+        results = []
+        restaurants = Restaurant.objects.none()
+
+        if Category.objects.filter(name__contains=search).exists():
+            categories = Category.objects.filter(name__contains=search)
+            for category in categories:
+                restaurants = Restaurant.objects.filter(category_id=category.id)
+
+        if Menu.objects.filter(item__contains=search).exists():
+            menus = Menu.objects.filter(item__contains=search)
+            for menu in menus:
+                restaurants = restaurants.union(Restaurant.objects.filter(id=menu.restaurant.id))
+
+        if Restaurant.objects.filter(address__contains=search):
+            restaurants = restaurants.union(Restaurant.objects.filter(address__contains=search))
+
+        if Restaurant.objects.filter(name__contains=search):
+            restaurants = restaurants.union(Restaurant.objects.filter(name__contains=search))
+
+        for restaurant in restaurants:
+            reviews = Review.objects.filter(restaurant_id=restaurant.id)
+            results.append(
+                {
+                    "id"          : restaurant.id,
+                    "name"        : restaurant.name,
+                    "address"     : restaurant.address,
+                    "is_wished"   : restaurant.wishlist_set.exists(),
+                    "rating"      : reviews.aggregate(Avg('rating')),
+                    "review"      : [{
+                    "description" : review.description,
+                    "image"       : [{
+                    "image_url"   : imageobject.image
+                } for imageobject in ReviewImage.objects.filter(review_id=review.id)]
+                } for review in reviews]
+                }
+            )
```

Comment:
아래와 같이 참조되어 있는 객체의 데이터는 부모 인스턴스로부터 가져올 수 있습니다. 아래와 같이 수정해주세요~

```suggestion
                {
                    "id"          : restaurant.id,
                    "name"        : restaurant.name,
                    "address"     : restaurant.address,
                    "is_wished"   : restaurant.wishlist_set.exists(),
                    "rating"      : reviews.aggregate(Avg('rating')),
                    "review"      : [{
                        "description" : review.description,
                        "images"      : [{"url" : image.url} for image in review.image_set.all()]
                    } for review in restaurant.review_set.all()]
                }
```

Reply:
큰 깨달음을 얻고 갑니다...